doc/python/performance.md

```diff
@@ -0,0 +1,298 @@
+* Available symbols will differ.
+* Area fills are not yet supported in WebGL.
+* Range breaks on time-series axes are not yet supported.
+* Axis range heuristics may differ.
+
+#### Multiple WebGL Contexts
+
+*New in 5.19*
+
+Most browsers have a limit of between 8 and 16 WebGL contexts per page. A Plotly WebGL-based figure may use multiple WebGL contexts, but generally you'll be able to render between 4 and 8 figures on one page.
+
+If you exceed the browser limit on WebGL contexts, some figures won't render and you'll see an error. In the console in Chrome, for example, you'll see the error: "Too many active WebGL contexts. Oldest context will be lost".
+
+If you encounter WebGL context limits when using WebGL-based figures, you can use [Virtual WebGL](https://github.com/greggman/virtual-webgl), which virtualizes a single WebGL context into multiple contexts.
+
+To use it, in the environment where your Plotly figures are being rendered, load the Virtual WebGL script, "https://unpkg.com/virtual-webgl@1.0.6/src/virtual-webgl.js", for example, using a `<script>` tag.
+
+In a Jupyter notebook environment that supports magic commands, you can load it with the [HTML magic command](https://ipython.readthedocs.io/en/stable/interactive/magics.html#cellmagic-html):
+
+
+```
+%%html
+<script src=“https://unpkg.com/virtual-webgl@1.0.6/src/virtual-webgl.js”></script>
+```
+
+
+### WebGL for Scatter Performance
+
+In the examples below we show that it is possible to represent up to around a million points with WebGL-enabled traces.
+For larger datasets, or for a clearer visualization of the density of points,
+it is also possible to use [datashader](/python/datashader/).
+
+### WebGL with Plotly Express
+
+The `render_mode` argument to supported Plotly Express functions (e.g. `scatter` and `scatter_polar`) can be used to enable WebGL rendering.
+
+> **Note** The default `render_mode` is `"auto"`, in which case Plotly Express will automatically set `render_mode="webgl"` if the input data is more than 1,000 rows long. In this case, WebGl can be disabled by setting `render_mode=svg`.
+
+Here is an example that creates a 100,000 point scatter plot using Plotly Express with WebGL rendering explicitly enabled.
+
+```python
+import plotly.express as px
+
+import pandas as pd
+import numpy as np
+
+np.random.seed(1)
+
+N = 100000
+
+df = pd.DataFrame(dict(x=np.random.randn(N),
+                       y=np.random.randn(N)))
+
+fig = px.scatter(df, x="x", y="y", render_mode='webgl')
+
+fig.update_traces(marker_line=dict(width=1, color='DarkSlateGray'))
+
+fig.show()
+```
+
+
+#### WebGL with 1,000,000 points with Graph Objects
+
+If Plotly Express does not provide a good starting point, it is also possible to use [the more generic `go.Scattergl` class from `plotly.graph_objects`](/python/graph-objects/).
```

Comment:
good starting point for what? (I know from context, but maybe worth repeating here?)